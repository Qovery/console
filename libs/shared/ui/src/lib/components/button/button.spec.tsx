import { render } from '@testing-library/react'

import Button from './button'

describe('Button', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Button />)
    expect(baseElement).toBeTruthy()
  })
})
