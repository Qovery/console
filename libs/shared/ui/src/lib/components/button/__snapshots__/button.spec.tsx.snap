// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-xs h-7 px-2 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 text-white disabled:text-brand-300 disabled:bg-brand-100"
      type="button"
    >
      Foobar
    </button>
  </div>
</body>
`;

exports[`Button should match snapshot with loading 1`] = `
<body>
  <div>
    <button
      class="inline-flex items-center shrink-0 font-medium transition focus-visible:[&:not(:active)]:outline-2 outline-0 shadow-sm disabled:shadow-none active:shadow-none outline-neutral-500 text-xs h-7 px-2 rounded bg-brand-500 active:bg-brand-600 hover:bg-brand-400 disabled:text-brand-300 disabled:bg-brand-100 pointer-events-none relative text-transparent"
      type="button"
    >
      <div
        class="aspect-square border-2  rounded-full border-solid border-neutral-350 border-r-neutral-50 animate-spin w-4 absolute left-0 right-0 m-auto"
        data-testid="spinner"
      />
      Foobar
    </button>
  </div>
</body>
`;
